crates/steel-parser/src/lexer.rs

```diff
@@ -597,14 +597,18 @@ pub struct OwnedTokenStream<'a, T, F> {
 }
 
 impl<'a, T, F: ToOwnedString<T>> Iterator for OwnedTokenStream<'a, T, F> {
-    type Item = Token<'a, T>;
+    type Item = std::result::Result<Token<'a, T>, TokenLike<'a, TokenError>>;
```

Comment:
:star: This is the main change: this iterator now returns a result that the parser must deal with.